Add InputEmailChip component

chiragbhaiji opened the pull request:
Fixes: #560 #557 

@sangameshsomawar please review

**Checklist**
- [x] I have performed a self-review of my code and tested it well before raising the PR.
- [x] I have made corresponding changes to the documentation and `index.d.ts`.
- [x] I've verified the change via .yalc in neetoDesk.  

**Demo or Screenshots**

Demo link: https://app.usebubbles.com/vJnXyiKBRMoGM1qF34jpMA/input-email-chip-component-demo

<img  width="320" src="https://user-images.githubusercontent.com/27726339/210410874-8ea49e2e-00c9-4fb2-bfad-4205872b5e10.png" />

sangameshsomawar commented on src/components/InputEmailChip.jsx:
```diff
@@ -0,0 +1,186 @@
+}) => {
+  const theme = useContext(ThemeContext);
+
+  const [emails, setEmails] = useState(initialEmails || []);
```
@chiragbhaiji 

https://app.usebubbles.com/cwEjgqgxQknywBJD8uwC1X/sangamesh-s-video-recording

chiragbhaiji commented on storybook/stories/Chip.stories.jsx:
```diff
@@ -17,7 +17,8 @@ const ChipStories = {
     variant: variants[0],
-    label: "Chip Component",
+    label:
+      "Chip ComponentChip ComponentChip ComponentChip ComponentChip ComponentChip Component",
```
Sorry I was testing the case of long strings after making updates to the Chip component. Will revert it.

chiragbhaiji commented:
@sangameshsomawar _a I've updated and fixed things based on the provided feedback. Please review the changes and check [this issue](https://app.usebubbles.com/rfPwpb8ckqfA9P7fVNAu1e/chirag-s-video-recording) as it requires your input for the fix.

sangameshsomawar reviewed (approved):
@chiragbhaiji _a LGTM, please test once and merge.

chiragbhaiji commented:
> @chiragbhaiji _a LGTM, please test once and merge.

Tested it on Pixel 4a emulator and it looks good on that too. So merged it.
